src/jit/emitxarch.cpp

```diff
@@ -5550,11 +5550,24 @@ static bool isSSEExtract(instruction ins)
     }
 }
 
+static bool isSseShuffle(instruction ins)
```

Comment:
I would change the name of this function, since it no longer encapsulates all SseShuffle, but instead only the shuffle instructions with a special encoding.

Reply:
This function can be called something like `noVexNdsThreeOpInstructions`

Reply:
I don't think that `noVexNdsThreeOpInstructions` is very clear. How about something like `isDstSrcImmSseInstruction` to correspond with `IsDstDstSrcAVXInstruction`?

Reply:
@CarolEidt `IsDstDstSrcAVXInstruction` implies the instruction has `VEX.NDS` encoding that duplicates the first-arg in the `VEX.vvvv` field. 
The problem that we are trying to solve here is some instructions ***do not*** have `VEX.NDS` encoding, which means they have the same register-form in SSE and VEX encoding. For example,
```asm
EXTRACTPS reg/m32, xmm1, imm8
VEXTRACTPS reg/m32, xmm1, imm8
```

So, I think the names corresponding to `IsDstDstSrcAVXInstruction` may be more confusing.

Reply:
How about `noVexNdsRegRegImmInstructions`?

Reply:
Considered again, `isDstSrcImmSseInstruction` is also good to me if we have some comments here 😄

Reply:
@CarolEidt @tannergooding @fiigii 

I have fixed this already by merging both `isSseExtract` and `isSseShuffle` functions. After consulting manual I have realized that `DstSrcImm` format is common to all ISAs including SSE, AVX and AVX512 and perhaps better name would be `isDstSrcImmSIMDInstruction`

Current implementation is as follows:

```C++
//------------------------------------------------------------------------
// isDstSrcImmSIMDInstruction: check if instruction has RM R I format
// for all encodings: EVEX, VEX and legacy SSE encoding
//
// Arguments:
//    instruction -- processor instruction to check
//
// Return Value:
//    true if instruction has RRI format
//
static bool isDstSrcImmSseInstruction(instruction ins)
{
    switch (ins)
    {
        case INS_extractps:
        case INS_pextrb:
        case INS_pextrw:
        case INS_pextrd:
        case INS_pextrq:
        case INS_pshufd:
        case INS_pshufhw:
        case INS_pshuflw:
            return true;
        default:
            return false;
    }
}
```

Reply:
@4creators, it probably just needs to be `IsDstSrcImmAvxInstruction`, as that clarifies that the VEX encoding is the same as the non-vex encoding (where it would otherwise get assumed to be `DstDstSrcImm` or `DstSrcSrcImm`)